improver/calibration/reliability_calibration.py

```diff
@@ -437,3 +437,74 @@ def process(self, historic_forecasts, truths):
             reliability_tables.append(reliability_entry)
 
         return MergeCubes()(reliability_tables)
+
+
+class AggregateReliabilityCalibrationTables:
+
+    """This plugin enables the aggregation of multiple reliability calibration
+    tables, and/or the aggregation over coordinates in the tables."""
+
+    def __init__(self):
+        """
+        Initialise plugin for manipulating reliability calibration tables.
+        This plugin can aggregate multiple tables or aggregate over coordinates
+        using summation.
+        """
+        self.cube_index_coord = 'cube_coord'
+        self.merge_coord = lambda value: iris.coords.DimCoord(
+            np.array([value]).astype(np.int32),
+            long_name=self.cube_index_coord, units=1)
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        return '<AggregateReliabilityCalibrationTables>'
+
+    def _construct_single_cube(self, cubes):
+        """
+        Add a new leading dimension to cubes and then use this to merge the
+        cubes into a single cube.
+
+        Args:
+            cubes (list or iris.cube.CubeList):
+                The cubes to be merged into a single cube.
+        Returns:
+            iris.cube.Cube:
+                A single cube containing all the input cubes with an additional
+                leading dimension.
+        """
+        merge_list = iris.cube.CubeList()
+        for index, cube in enumerate(cubes):
+            local_cube = cube.copy()
+            local_cube.add_aux_coord(self.merge_coord(index))
+            merge_list.append(
+                iris.util.new_axis(local_cube, self.cube_index_coord))
+        cube = merge_list.concatenate()
+        return cube.merge_cube()
+
+    def process(self, cubes, coordinates=None):
+        """
+        Aggregate the input reliability calibration table cubes and return the
+        result.
+
+        Args:
+            cubes (list or iris.cube.CubeList):
+                The cube or cubes containing the reliability calibration tables
+                to aggregate.
+            coordinates (list or None):
+                A list of coordinates over which to aggregate the reliability
+                calibration table using summation. If the argument is None and
+                a single cube is provided, this cube will be returned
+                unchanged.
+        """
+        coordinates = [] if coordinates is None else coordinates
+
+        try:
+            cube, = cubes
+        except ValueError:
+            cube = self._construct_single_cube(cubes)
+            coordinates.append(self.cube_index_coord)
+        else:
+            if not coordinates:
+                return cube
+
+        return collapsed(cube, coordinates, iris.analysis.SUM)
```

Comment:
At the moment, the `cube_coord` is still present on the output cube but, maybe, as this is a coordinate that doesn't have any meaning outside of this plugin, then this coordinate should be removed. What do you think?